### PR TITLE
Implement True Positive Writebacks

### PR DESCRIPTION
**Pull request by jeberl**
Summary
---------

Implement True positive Writbacks to ThreatExchange. This will upsert a new descriptor for the given indicator with:
1. MALICIOUS
2. The same privacy group
3. MANUALLY REVIEWED

If the descriptor does not already exist, the following feilds will also be set:
1. Description = "A ThreatDescriptor uploaded via HMA"
2. Tags = "uploaded_by_hma"
3. share level = "RED

If the descriptor already exists, those fields will be unchanged

Test Plan
---------

1. Created new test application (HMA test)
2. Uploaded all test data threat descriptors again with the new app
3. Deleted old Threat Descriptors from the old testing app (Jesse's terrorist stopping app)
4. Triggered TruePositive writeback locally
5. Watched new ThreatDescriptor get created in ThreatExchange

### CODE REVIEW

File: hasher-matcher-actioner/hmalib/common/writebacker_models.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/common/writebacker_models.py:244]: nit: confusing variable name logic flow here

File: hasher-matcher-actioner/hmalib/common/writebacker_models.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/common/writebacker_models.py:256]: what if `privacy_members` (i.e. `privacy_group_id` see nit about confusing name above) is in members already? I'm not sure but I think it is possible that depending on what copy does for existing descriptor that this will remove it from the existing other privacy groups...

**Review by BarrettOlson — CHANGES_REQUESTED** [review]
Generally looks pretty good but flagged some blocking concerns inline to follow up on before merge. Thanks!

File: hasher-matcher-actioner/hmalib/common/writebacker_models.py
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/common/writebacker_models.py:238]: Yeah. Its like an upsert

Comment (BarrettOlson) [reply hasher-matcher-actioner/hmalib/common/writebacker_models.py:238]: cool cool, nit: can update the comment to reflect that?

File: hasher-matcher-actioner/hmalib/common/writebacker_models.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/common/writebacker_models.py:266]: why RED vs AMBER?

File: hasher-matcher-actioner/hmalib/common/writebacker_models.py
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/common/writebacker_models.py:27]: Sure

Comment (BarrettOlson) [reply hasher-matcher-actioner/hmalib/common/writebacker_models.py:27]: Sure!

File: hasher-matcher-actioner/hmalib/lambdas/actions/writebacker.py
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/lambdas/actions/writebacker.py:48]: `MOCK_TE_API` is a different flag for unit tests. 

This pattern is pretty similar to those in other `__name__ == "__main__"` blocks such as in pdq_indexer. For basic debugging. I can document to make it clearer and if you feel strongly can add another environ flag

File: hasher-matcher-actioner/hmalib/common/writebacker_models.py
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/common/writebacker_models.py:241]: Yeah your right. Ran into this in a stacked diff. Haven't tested this case e2e yet.

File: hasher-matcher-actioner/hmalib/common/writebacker_models.py
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/common/writebacker_models.py:266]: I think stricter is probably better for a default. Something we could make customizable later

File: hasher-matcher-actioner/hmalib/lambdas/actions/writebacker.py
Comment (BarrettOlson) [reply hasher-matcher-actioner/hmalib/lambdas/actions/writebacker.py:48]: Right, that flag could be set here (or even treated as a default) because we are reacting to real descriptors better documenting and preferably another level of confirmation before doing so would be ideal imo. (e.g. using https://docs.python.org/3/library/functions.html#input)

File: hasher-matcher-actioner/hmalib/common/writebacker_models.py
Comment (BarrettOlson) [reply hasher-matcher-actioner/hmalib/common/writebacker_models.py:170]: Did you update this? (don't see a change)

File: hasher-matcher-actioner/hmalib/common/writebacker_models.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/common/writebacker_models.py:183]: rm or change to debug log

File: hasher-matcher-actioner/hmalib/lambdas/actions/writebacker.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/lambdas/actions/writebacker.py:49]: Thanks for adding this! if you want to make it even more idiot proof for someone like me trying to use it later you could also add (probably want to test to make sure)
`# example usage: $ WRITEBACK_LOCAL=True python3 -m hmalib/lambdas/actions/writebaker.py`

**Review by BarrettOlson — APPROVED** [review]
Thanks for making the changes and bug fixes! Stamping to unblock. Did you verify that the changes made by the write-backer result in necessary updates to the .te files or python-threatexchange yet? (needed but can be in the next PR imo)

**Comment by jeberl** [comment]
Coming in a later pr

**Pull request merged**
